app/scripts/controllers/serviceInstance.js

```diff
@@ -73,18 +73,40 @@ angular.module('openshiftConsole')
       DataService.unwatchAll(secretWatchers);
       secretWatchers = [];
 
+      $scope.allowParametersReveal = AuthorizationService.canI('secrets', 'get', $scope.projectName);
       $scope.parameterData = {};
-      _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function(key) {
-        $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
+      $scope.opaqueParameterKeys = [];
+
+      if (!$scope.allowParametersReveal) {
+        _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function (key) {
+          $scope.parameterData[key] = '*****';
+        });
+      }
+
+      var statusParameters = _.get($scope.serviceInstance, 'status.externalProperties.parameters', {});
+      _.each(_.keys(statusParameters), function(key) {
+        if (statusParameters[key] === '<redacted>') {
+          $scope.parameterData[key] = '*****';
+        } else {
+          $scope.parameterData[key] = statusParameters[key];
+          $scope.opaqueParameterKeys.push(key);
+        }
       });
 
-      $scope.parameterData = angular.extend($scope.parameterData, _.get($scope.serviceInstance, 'spec.parameters', {}));
+      // Fill in the secret values if they are available to the user
+      if ($scope.allowParametersReveal) {
+        // Set default values for those items that are not specifically set, secrets will override
+        _.each(_.keys(_.get($scope.parameterSchema, 'properties')), function (key) {
+          $scope.parameterData[key] = $scope.parameterSchema.properties[key].default;
```

Comment:
I think the defaults are only for prefiling the create form. A missing value isn't necessarily treated as a default.

Reply:
Fixed.